source/MaterialXGenShader/Util.cpp

```diff
@@ -363,78 +363,62 @@ bool elementRequiresShading(ConstTypedElementPtr element)
     return colorClosures.count(elementType) > 0;
 }
 
-void findRenderableMaterialNodes(ConstDocumentPtr doc,
-                                 vector<TypedElementPtr>& elements,
-                                 bool includeReferencedGraphs,
-                                 std::unordered_set<ElementPtr>& processedSources)
+vector<TypedElementPtr> findRenderableMaterialNodes(ConstDocumentPtr doc)
 {
-    for (const NodePtr& material : doc->getMaterialNodes())
+    vector<TypedElementPtr> renderableNodes;
+    for (NodePtr materialNode : doc->getMaterialNodes())
     {
-        // Scan for any upstream shader outputs and put them on the "processed" list
-        // if we don't want to consider them for rendering.
-        vector<NodePtr> shaderNodes = getShaderNodes(material);
-        if (!shaderNodes.empty())
+        if (!getShaderNodes(materialNode).empty())
         {
-            // Push the material node only once if any shader nodes are found
-            elements.push_back(material);
-            processedSources.insert(material);
-
-            if (!includeReferencedGraphs)
-            {
-                for (NodePtr shaderNode : shaderNodes)
-                {
-                    for (InputPtr input : shaderNode->getActiveInputs())
-                    {
-                        OutputPtr outputPtr = input->getConnectedOutput();
-                        if (outputPtr && !outputPtr->hasSourceUri() && !processedSources.count(outputPtr))
-                        {
-                            processedSources.insert(outputPtr);
-                        }
-                    }
-                }
-            }
+            renderableNodes.push_back(materialNode);
         }
     }
+    return renderableNodes;
 }
 
-void findRenderableElements(ConstDocumentPtr doc, vector<TypedElementPtr>& elements, bool includeReferencedGraphs)
+vector<TypedElementPtr> findRenderableElements(ConstDocumentPtr doc)
 {
-    std::unordered_set<ElementPtr> processedSources;
-    findRenderableMaterialNodes(doc, elements, includeReferencedGraphs, processedSources);
-    if (!elements.empty())
+    vector<TypedElementPtr> renderableElements = findRenderableMaterialNodes(doc);
```

Comment:
This is missing finding shader nodes not connected to materials at the top level, but I believe this was "lost" in a previous change.

Reply:
Good point, though it looks like we've lacked this feature since the transition from 1.37 to 1.38, so I'd like to make sure it's still required for adding the functionality back.